optimum/habana/utils.py

```diff
@@ -216,3 +216,69 @@ def get_driver_version():
     if output.returncode == 0:
         return version.parse(output.stdout.split("\n")[2].replace(" ", "").split(":")[1][:-1].split("-")[0])
     return None
+
+
+class HabanaProfile(object):
+    """
+    HPU profiler only could be run once, so HABANA_PROFILE_ENABLED, a class static variable shared by all the instances of HabanaProfile, is used to control which part will be captured
+    """
+
+    HABANA_PROFILE_ENABLED = False
+
+    def __init__(self, warmup: int = 0, active: int = 0, output_dir: str = "./hpu_profile", wait: int = 0):
+        if active <= 0 or warmup <= 0 or HabanaProfile.HABANA_PROFILE_ENABLED:
+
+            def noop():
+                pass
+
+            self.start = noop
+            self.stop = noop
+            self.step = noop
+        else:
+            HabanaProfile.HABANA_PROFILE_ENABLED = True
+            schedule = torch.profiler.schedule(wait=wait, warmup=warmup, active=active, repeat=1)
+            activities = [torch.profiler.ProfilerActivity.CPU, torch.profiler.ProfilerActivity.HPU]
+
+            profiler = torch.profiler.profile(
+                schedule=schedule,
+                activities=activities,
+                on_trace_ready=torch.profiler.tensorboard_trace_handler(output_dir),
+                record_shapes=True,
+                with_stack=True,
+            )
+            self.start = profiler.start
+            self.stop = profiler.stop
+            self.step = profiler.step
+            HabanaProfile.enable.invalid = True
+            HabanaProfile.disable.invalid = True
```

Comment:
What do these attributes do?

Reply:
what I want is:
- `enable` and `disable` could run many times before `return a profiler object to user`
- `enable` and `disable` should not work after   `return a profiler object to user`

that's because in one process we only can do profiling once, after we generate the profiling, we can not enable it again.